### PR TITLE
Refactored date picker - pulled changes from the last PR

### PR DESCRIPTION
**Pull request by YanivWein24**

### CODE REVIEW

**Comment by SereneSound** [comment]
@YanivWein24 Why are there so many changes here, did you run the linter on this?

File: src/pages/dogs/ReportDogPage.tsx [hunk 1]
```diff
@@ -17,310 +19,338 @@ import { RTLTextField } from "../../components/pageComponents/RTLTextInput/RTLTe
...
-    if (dogType === DogType.LOST) {
-      return AppTexts.reportPage.title.lost;
-    }
-    return AppTexts.reportPage.title.found;
-  };
+                const { dateInput } = inputs.date;
+                // @ts-expect-error
+                const { $D, $M, $y } = dateInput;
+                // format the selected date to match dd/mm/yyyy
+                return `${withZero($D)}/${withZero($M)}/${$y}`;
```
Comment (SereneSound) [inline src/pages/dogs/ReportDogPage.tsx:132]: Yaniv David requested the format `YYYY-MM-DD` in the backend

**Review by SereneSound — CHANGES_REQUESTED** [review]

**Comment by YanivWein24** [comment]
> @YanivWein24 Why are there so many changes here, did you run the linter on this?

@SereneSound We don't have a linter configured in this branch, but I do have the Prettier extension in VSC.
The reason for the large number of changes is that inside the ".editorconfig" file we configured: "indent_size = 4" (even though we use 2 spaces per indent in this project).
Prettier extension has read this file and changed the indentation size on every file I have worked on.
If you wish, I can change the "indent_size" to 2 and push it again. This time it will push it without the added spaces.

Other than that, I also had to create `useDateInput` hook and install `Dayjs` because this is what MUI uses under the hood in their date-picker component (and it doesn't work with normal Date())

**Review by SereneSound — APPROVED** [review]
@YanivWein24 Can you resolve conflicts so we merge this?

**Pull request merged**